lib/vanagon/component.rb

```diff
@@ -350,17 +350,18 @@ def extract_with # rubocop:disable Lint/DuplicateMethods
 
     # Fetches patches if any are provided for the project.
     #
-    # @param workdir [String] working directory to put the patches into
-    def get_patches(workdir)
-      unless @patches.empty?
-        patchdir = File.join(workdir, "patches")
-        FileUtils.mkdir_p(patchdir)
-        @patches.each do |p|
-          target = File.join(patchdir, File.basename(p.path))
-          raise Vanagon::Error, "Duplicate patch files detected, '#{p.path}' would have overwritten '#{target}'! Please ensure patch file names are unique across all components." if File.exist?(target)
-
-          FileUtils.cp(p.path, patchdir)
+    # @param patch_root [String] working directory to put the patches into
+    def get_patches(patch_root)
+      return if @patches.empty?
+      @patches.each do |patch|
+        patch_assembly_path = File.join(patch_root, patch.assembly_path)
+        if File.exist?(patch_assembly_path)
+          raise Vanagon::Error,
+                "Duplicate patch files detected, '#{patch.origin_path}' would have overwritten '#{patch.assembly_path}'"
```

Comment:
I like the "Please ensure patch file names are unique across all components." part of the previously-existing error message, since it gives you a push in the right direction for what needs to be changed. Could that be added back here?

Reply:
Good idea but I'm struggling with the wording. Maybe:

"Ensure all patch file names within a component are unique"

(Also, I have a bugaboo that computers are not allowed 'please' and 'thank you'. That is reserved strictly for human interaction. However, I can keep it here if that sort of consistency is desirable).

Reply:
I think that works! Or perhaps "Ensure unique file names for all patches of a given component."?